cni/network/invoker_azure.go

```diff
@@ -58,20 +56,21 @@ func (invoker *AzureIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, er
 	}
 
 	// Call into IPAM plugin to allocate an address pool for the network.
-	addResult.ipv4Result, err = invoker.plugin.DelegateAdd(addConfig.nwCfg.IPAM.Type, addConfig.nwCfg)
-
+	ipv4Result, err := invoker.plugin.DelegateAdd(addConfig.nwCfg.IPAM.Type, addConfig.nwCfg)
 	if err != nil && strings.Contains(err.Error(), ipam.ErrNoAvailableAddressPools.Error()) {
 		invoker.deleteIpamState()
 	}
 	if err != nil {
 		err = invoker.plugin.Errorf("Failed to allocate pool: %v", err)
 		return addResult, err
 	}
+	addResult.defaultCniResult = CNIResult{ipResult: ipv4Result, addressType: cns.Default, isDefaultInterface: true}
+	addResult.hostSubnetPrefix = ipv4Result.IPs[0].Address
```

Comment:
there's no change here, just moved this line from end of func

Reply:
Yes but I agree with Vipul you could get index out of range even in the previous code, if ipv4result.IPs is nil

Reply:
it checks on the defer because if err != nil then the result from DelegateAdd may be nil, but in this part, we return the err if it's not nil and access the result otherwise

Reply:
ok we can add check